src/components/shared/Fields/Form/HookForm.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import {
+  useForm,
+  UseFormProps,
+  FormProvider,
+  UseFormReturn,
+  SubmitHandler,
+  SubmitErrorHandler,
+} from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Schema } from 'yup';
+
+const displayName = 'HookForm';
+
+export interface HookFormProps<FormData extends Record<string, any>> {
+  children: ((props: UseFormReturn) => React.ReactNode) | React.ReactNode;
+  onSubmit: SubmitHandler<FormData>;
+  onError?: SubmitErrorHandler<FormData>;
+  validationSchema?: Schema<FormData>;
+  defaultValues?: UseFormProps['defaultValues'];
+  mode?: UseFormProps['mode'];
+  options?: UseFormProps;
+}
+
+const HookForm = <FormData extends Record<string, any>>({
+  children,
+  defaultValues,
+  mode = 'onTouched',
+  onSubmit,
+  onError,
+  options,
+  validationSchema,
+}: HookFormProps<FormData>) => {
+  const formHelpers = useForm({
+    resolver: validationSchema ? yupResolver(validationSchema) : undefined,
+    defaultValues,
+    mode,
+    ...options,
+  });
+  const { handleSubmit } = formHelpers;
+  return (
+    <FormProvider {...formHelpers}>
+      <form onSubmit={handleSubmit(onSubmit, onError)}>
+        {typeof children === 'function' ? children(formHelpers) : children}
+      </form>
+    </FormProvider>
+  );
+};
+
+HookForm.displayName = displayName;
```

Comment:
Modelled on the style of our Formik form, this wraps a form element in the React Hook Form context and passes in the helpers to a child function that returns JSX, render-props style.